internal/configs/version1/template_helper.go

```diff
@@ -64,14 +66,162 @@ func makePathWithRegex(path, regexType string) string {
 	}
 }
 
+var setHeader = regexp.MustCompile("[a-zA-Z]+$")
+
+func validateProxySetHeader(header string) error {
+	header = strings.TrimSpace(header)
+	if !setHeader.MatchString(header) {
+		return errors.New("invalid header syntax")
+	}
+	return nil
+}
+
+func defaultHeaderValues(headerParts []string, headerName string) string {
+	headerValue := strings.TrimSpace(headerParts[0])
+	headerValue = strings.ReplaceAll(headerValue, "-", "_")
+	headerValue = strings.ToLower(headerValue)
+	return fmt.Sprintf("\n\t\tproxy_set_header %s $http_%s;", headerName, headerValue)
+}
+
+func headersGreaterThanOne(headerParts []string, header string, headerName string) (string, error) {
+	headerValue := strings.TrimSpace(headerParts[1])
+	if strings.Contains(headerValue, " ") {
+		return "", errors.New("multiple values found in header: " + header)
+	}
+	return fmt.Sprintf("\n\t\tproxy_set_header %s %q;", headerName, headerValue), nil
+}
+
+func splitHeaders(header string) (string, []string, string) {
+	header = strings.TrimSpace(header)
+	headerParts := strings.SplitN(header, " ", 2)
+	headerName := strings.TrimSpace(headerParts[0])
+	return header, headerParts, headerName
+}
+
+// minionProxySetHeaders takes an ingress annotations map and a bool map
+// and generates proxy_set_header headers for minions based on the nginx.org/proxy-set-headers annotation in a mergable ingress.
+// It returns a string containing the generated proxy headers and a header.
+func minionProxySetHeaders(loc *Location, minionHeaders map[string]bool) (string, map[string]bool, error) {
+	proxySetHeaders, ok := loc.MinionIngress.Annotations["nginx.org/proxy-set-headers"]
+	if !ok {
+		return "", nil, nil
+	}
+	var combinedMinions string
+	headers := strings.Split(proxySetHeaders, ",")
+	for _, header := range headers {
```

Comment:
can this for loop be a function?

Reply:
I have tried and for the moment, that doesn't seem possible.